code/mob/living/critter/aberration.dm

```diff
@@ -0,0 +1,155 @@
+	speechverb_exclaim = "forces"
+	speechverb_stammer = "creates"
+	speechverb_gasp = "rasps"
+	speech_void = TRUE
+
+	can_burn = FALSE
+	can_implant = FALSE
+	canbegrabbed = FALSE
+	throws_can_hit_me = FALSE
+	reagent_capacity = 0
+	caneat = FALSE
+	candrink = FALSE
+	blood_id = null
+	can_bleed = FALSE
+	metabolizes = FALSE
+	use_stamina = FALSE
+
+	grabresistmessage = "but their hands pass right through %src%!"
+	death_text = "%src% dissipates!"
+
+	New()
+		..()
+
+		remove_lifeprocess(/datum/lifeprocess/blood)
+		remove_lifeprocess(/datum/lifeprocess/chems)
+		remove_lifeprocess(/datum/lifeprocess/fire)
+		remove_lifeprocess(/datum/lifeprocess/mutations)
+		remove_lifeprocess(/datum/lifeprocess/organs)
+		remove_lifeprocess(/datum/lifeprocess/skin)
+		remove_lifeprocess(/datum/lifeprocess/stuns_lying)
+		remove_lifeprocess(/datum/lifeprocess/viruses)
+		remove_lifeprocess(/datum/lifeprocess/blindness)
+		remove_lifeprocess(/datum/lifeprocess/radiation)
+
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_RADPROT_INT, src, 100)
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_RADPROT_EXT, src, 100)
+		APPLY_ATOM_PROPERTY(src, PROP_ATOM_FLOATING, src)
+		APPLY_ATOM_PROPERTY(src, PROP_MOB_NIGHTVISION, src)
+
+	death()
+		..()
+		qdel(src)
+
+	setup_hands()
+		..()
+		var/datum/handHolder/HH = hands[1]
+		HH.limb = new /datum/limb/aberration_field
+		HH.name = "particle field"
+		HH.limb_name = HH.name
+		HH.can_hold_items = FALSE
+
+	setup_healths()
+		var/datum/healthHolder/brute = src.add_health_holder(/datum/healthHolder/aberration)
+		brute.value = 100
+		brute.maximum_value = 100
+		brute.last_value = 100
+
+	attack_hand(mob/living/M, params, location, control)
+		boutput(M, "<span class='combat'><b>Your hand passes right through! It's so cold...</b></span>")
+		return
+
+	attackby(obj/item/I, mob/M)
+		if (!istype(I, /obj/item/baton))
+			boutput(M, "<span class='combat'><b>[I] passes right through!</b></span>")
+			return
+
+		var/obj/item/baton/B = I
+		if (!B.can_stun(1, M))
+			return
+		M.visible_message("<span class='combat'><b>[M] shocks the [src.name] with [I]!</b></span>",
+			"<span class='combat'><b>While your baton passes through, the [src.name] appears damaged!</b></span>")
+		M.lastattacked = src
+		B.process_charges(-1, M)
+
+		src.hurt(50)
+
+	bullet_act(obj/projectile/P)
+		if (P.proj_data.damage_type == D_ENERGY && round(P.power * (1 - P.proj_data.ks_ratio), 1) > 1)
+			src.hurt(100)
+
+	projCanHit(datum/projectile/P)
+		return P.damage_type == D_ENERGY
+
+	do_disorient(stamina_damage, weakened, stunned, paralysis, disorient = 60, remove_stamina_below_zero = 0, target_type = DISORIENT_BODY, stack_stuns = 1)
+		return
+
+	TakeDamage(zone, brute, burn, tox, damage_type, disallow_limb_loss)
+		return // no healing
+
+	ex_act(severity)
+		return
+
+	blob_act(power)
+		return
+
+	proc/hurt(damage)
+		var/datum/healthHolder/Br = src.get_health_holder("brute")
+		Br?.TakeDamage(damage)
+
+	seek_target(range)
+		var/targets = ..(7)
```

Comment:
looping through twice seems inefficient. It'd be better to just copy the parent proc and add an extra continue for the crown.